pkg/util/docker/containers.go

```diff
@@ -182,6 +184,58 @@ func parseContainerHealth(status string) string {
 	return all[0][1]
 }
 
+// parseContainerNetworkAddresses converts docker ports
+// and network settings into a list of NetworkAddress
+func parseContainerNetworkAddresses(ports []types.Port, netSettings *types.SummaryNetworkSettings, container string) []containers.NetworkAddress {
+	addrList := []containers.NetworkAddress{}
+	tempAddrList := []containers.NetworkAddress{}
+	if netSettings == nil || len(netSettings.Networks) == 0 {
+		log.Debugf("No network settings available from docker")
+		return addrList
+	}
+	for _, port := range ports {
+		if isExposed(port) {
+			IP := net.ParseIP(port.IP)
+			if IP == nil {
+				log.Warnf("Unable to parse IP: %v for container: %s", port.IP, container)
+				continue
+			}
+			addrList = append(addrList, containers.NetworkAddress{
+				IP:       IP,                   // Host IP, since the port is exposed
+				Port:     int(port.PublicPort), // Exposed port
+				Protocol: port.Type,
+			})
+		} else {
```

Comment:
if it is not exposed why do we want to list them?

Reply:
because they still reachable by other containers in same network

Reply:
I'm wondering if in kubernetes environment inside a Pod, containers can target other container port without be listed by docker as Public/Private port.

Reply:
I see what you mean here, actually a PR for the same purpose (collecting network info for network tracer) has been made for kubelet https://github.com/DataDog/datadog-agent/pull/2664 so I guess process team should rely on it when we're using docker + kubernetes